src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -780,10 +782,12 @@ protected FieldValueInfo completeValueForList(ExecutionContext executionContext,
 
         List<FieldValueInfo> fieldValueInfos = new ArrayList<>(size.orElse(1));
         int index = 0;
-        for (Object item : iterableValues) {
+        Iterator<Object> iterator = iterableValues.iterator();
```

Comment:
Sorry why do this ?

Reply:
To check incrementAndCheckMaxNodesExceeded before its exceeded ?

Reply:
reverted